File: InventoryManagement.Core/Entities/PurchaseOrder.cs
using InventoryManagement.Core.Common;
using InventoryManagement.Core.Enums;

namespace InventoryManagement.Core.Entities;

public class PurchaseOrder : BaseEntity
{
    public string OrderNumber { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; }
    public int VendorId { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public decimal TotalAmount { get; set; }
    public decimal? TaxAmount { get; set; }
    public decimal? DiscountAmount { get; set; }
    public decimal GrandTotal { get; set; }
    public string? Notes { get; set; }
    public DateTime? ExpectedDeliveryDate { get; set; }
    public DateTime? ActualDeliveryDate { get; set; }

    // Navigation Properties
    public Vendor Vendor { get; set; } = null!;
    public ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; } = new List<PurchaseOrderItem>();
}

File: InventoryManagement.UI/src/app/core/models/purchase-order.ts
export enum OrderStatus {
  Draft = 'Draft',
  Pending = 'Pending',
  Confirmed = 'Confirmed',
  Received = 'Received',
  Cancelled = 'Cancelled'
}

export interface PurchaseOrder {
  id: number;
  orderNumber: string;
  orderDate: Date;
  vendorId: number;
  vendorName: string;
  status: OrderStatus;
  totalAmount: number;
  taxAmount?: number;
  discountAmount?: number;
  grandTotal: number;
  notes?: string;
  expectedDeliveryDate?: Date;
  actualDeliveryDate?: Date;
  createdAt: Date;
  items: PurchaseOrderItem[];
}

export interface PurchaseOrderItem {
  id: number;
  itemId: number;
  itemName: string;
  itemCode: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  discountPercent?: number;
  discountAmount?: number;
  netAmount: number;
}

export interface CreatePurchaseOrderRequest {
  orderDate: Date;
  vendorId: number;
  taxAmount?: number;
  discountAmount?: number;
  notes?: string;
  expectedDeliveryDate?: Date;
  items: CreatePurchaseOrderItemRequest[];
}

export interface CreatePurchaseOrderItemRequest {
  itemId: number;
  quantity: number;
  unitPrice: number;
  discountPercent?: number;
}

export interface UpdatePurchaseOrderRequest {
  orderDate: Date;
  vendorId: number;
  taxAmount?: number;
  discountAmount?: number;
  notes?: string;
  expectedDeliveryDate?: Date;
  items: CreatePurchaseOrderItemRequest[];
}

export interface PagedPurchaseOrderResult {
  items: PurchaseOrder[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}